greedy/homework6.tex

```diff
@@ -0,0 +1,225 @@
+\documentclass[12pt]{article}
+
+\usepackage{tikz}
+\usetikzlibrary{trees}
+\usepackage[]{algorithm2e}
+\usepackage{amsmath}
+\newcommand{\BigO}[1]{\ensuremath{\operatorname{O}\bigl(#1\bigr)}}
+
+\begin{document}
+\title{Homework 5}
+\author{Robbie McKinstry, Jack McQuown, Cyrus Ramavarapu}
+\renewcommand{\today}{12 September 2016}
+\renewcommand{\baselinestretch}{1.5}
+
+\maketitle
+
+\section*{Greedy Problems}
+\subsection*{Problem 7:}
+A greedy algorithm that will minimize evictions is to 
+select the page that will be accessed furthest in the future
+from the current time.\\\\
+For example, assuming $k=4$ let the current state of the system be the
+following:
+\begin{center}
+    \begin{tabular}{c|c|c|c|c|c|c|c}
+    Time & 1 & 2 & 3 & 4 & 5 & 6 & 7 \\
+    Input & A & C & D & A & B & A & A \\   
+    \end{tabular}
+    Fast Memory ($k=4$) 
+    \begin{tabular}{c|c|c|c}
+    E & D & F & B 
+    \end{tabular}
+\end{center}
+In this situation, the algorithm will evict page B, since
+it is not used until time 5.  Any other choice will result
+in an earlier eviction. 
+\subsubsection*{Proof by Exchange:}
+Let $Alg$ be the process by which the above algorithm operates.
+Assume there exists an input $I$ such that $Alg(I)$ is 
+incorrect.  Let $Opt(I)$ be the optimal result for $I$
+that agrees with the greatest number of steps with $Alg(I)$.
+$Alg(I)$ and $Opt(I)$ must have a first point of disagreement.
+Label this time $t_i$.\\\\
+At time $i$, $Alg(I)$ must have evicted some page $u$ and 
+$Opt(I)$ has some page $v$ evicted.  Since $Alg(I)$ always
+picks the page that will be used furthest in the future,
+the next time page $u$ will be used must be further into
+the future than the next time page $v$ is used.  Let these
+times be respectively $t_u$ and $t_v$.\\\\
+Since $Alg(I)$ and $Opt(I)$ both agreed upto time $t_i$,
+pages $u$ and $v$ must be in the fast memory of $Opt(I)$
+at $t_i$.  Therefore, define $Opt'(I)$ as $Opt(I)$ except
+at time $t_i$ evict page $u$ instead of $v$.\\\\
+To show that $Opt'(I)$ is at least as optimal as $Opt(I)$     
+it needs to be recognized that since $t_u > t_v$, page $v$
+will have to be brought back into memory at least $1$ more
+time before page $u$ is brought back into memory.\\\\
+Since $Opt'(I)$ is at least as optimal as $Opt(I)$ and 
```

Comment:
Added a pargraph after the timing discussion with your point.  Tried to make them flow together logically, since the timing info is connected to the eviction count.